app/main.py

```diff
@@ -1,3 +1,26 @@
 def cache(func):
-    # Write your code here
-    pass
+    caches = []
+
+    def inner(*args):
+        local_cache = []
+        for arg in args:
+            local_cache.append(arg)
+        if local_cache in caches:
+            print("Getting from cache")
+        else:
+            print("Calculating new result")
+            caches.append(local_cache)
+
+        return func(*args)
```

Comment:
your approach is incorrect: you should store arguments and the result of the function with these arguments and return this result without calling function when arguments are the same.
hint: use dictionary as storage